src/views/Generator/Thresholds/ThresholdsDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import { ThresholdDataSchema } from '@/schemas/generator'
+import { useGeneratorStore } from '@/store/generator'
+import { Threshold, ThresholdData } from '@/types/thresholds'
+import { css } from '@emotion/react'
+import { zodResolver } from '@hookform/resolvers/zod'
+import {
+  Dialog,
+  Table,
+  Button,
+  Text,
+  Link as RadixLink,
+} from '@radix-ui/themes'
+import { useCallback, useEffect } from 'react'
+import { useForm, useFieldArray, FormProvider } from 'react-hook-form'
+import { ThresholdRow } from './ThresholdRow'
+import { uniqueId } from 'lodash-es'
+
+type ThresholdsDialogProps = {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+}
+
+export function ThresholdsDialog({
+  open,
+  onOpenChange,
+}: ThresholdsDialogProps) {
+  const thresholds = useGeneratorStore((store) => store.thresholds)
+
+  const formMethods = useForm<{ thresholds: Threshold[] }>({
+    resolver: zodResolver(ThresholdDataSchema),
+    shouldFocusError: false,
+    defaultValues: {
+      thresholds,
+    },
+  })
+
+  const { handleSubmit, control, watch } = formMethods
+
+  const handleOpenDocs = () =>
+    window.studio.browser.openExternalLink(
+      'https://grafana.com/docs/k6/latest/using-k6/thresholds/'
+    )
+
+  const { append, remove, fields } = useFieldArray<ThresholdData>({
+    control,
+    name: 'thresholds',
+  })
+
+  function handleAddThreshold(event: React.MouseEvent) {
+    event.preventDefault()
+
+    append({
+      id: uniqueId(),
+      url: '*',
+      metric: 'http_req_duration',
+      statistic: 'avg',
+      condition: '>',
+      value: 0,
+      stopTest: false,
+    })
+  }
+
+  const onSubmit = useCallback((data: ThresholdData) => {
+    console.log('onSubmit', data)
+  }, [])
+
+  // Submit onChange
+  useEffect(() => {
+    const subscription = watch(() => handleSubmit(onSubmit)())
+    return () => subscription.unsubscribe()
+  }, [watch, handleSubmit, onSubmit])
+
+  return (
+    <Dialog.Root open={open} onOpenChange={onOpenChange}>
+      <Dialog.Content
```

Comment:
I'm using a dialog for now so I can get the remaining UI working. Any feedback is appreciated here if you think this isn't the best approach.